src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -52,7 +52,7 @@ public partial class DynamoModel : INotifyPropertyChanged, IDisposable, IEngineC
         private readonly PathManager pathManager;
         private WorkspaceModel currentWorkspace;
         private Timer backupFilesTimer;
-
+        private Dictionary<Guid, string> backupFilesDict = new Dictionary<Guid, string>();
```

Comment:
Hi @Randy-Ma, I think I understand your points about needing this so that we don't generate new file names for existing workspaces that have already been backed-up, but this `backupFilesDict` can be a local variable in `SaveBackupFiles` method. So the whole thing in `SaveBackupFiles` works like this:

```csharp
OnRequestDispatcherBeginInvoke(() =>
{
    // This is now a local variable...
    var backupFilesDict = PreferenceSettings.BackupFiles;
    var tempDict = new Dictionary<Guid,string>(backupFilesDict);
    PreferenceSettings.BackupFiles.Clear();
    
    // Bla bla bla...
    
    // Adding this back to the permanent storage. Of course this will require 
    // "PreferenceSettings.BackupFiles" to be a "Dictionary" also, if that is 
    // okay. The only complication for this is "Dictionary" cannot be serialized 
    // into XML file natively (there needs to be some additional codes for that).
    // 
    PreferenceSettings.BackupFiles.AddRange(backupFilesDict);
});
```

The reason I am still proposing this is that I'd prefer not to have both `PreferenceSettings` and `DynamoModel` managing the back-up related data. The more this can be contained in just `PreferenceSettings` the better, but that's just my personal opinion about not distributing the data.

Please let me know what your thoughts are regarding this.